src/routes/api/product.routes.js

```diff
@@ -0,0 +1,30 @@
+import express from 'express';
+import { protectRoute, restrictTo } from '../../middlewares/auth.middleware';
+import { createProduct } from '../../controllers/product.controller.js';
+import productValidation from '../../validations/product/product.validation';
+import { addCategory } from '../../controllers/category.controller';
+import { checkIfCategoryExists, checkCategory } from '../../middlewares/category.middleware';
+import { checkExtensions, checkIfProductExists, checkNumberOfImages, uploadImages } from '../../middlewares/product.middleware';
+import categoryValidation from '../../validations/product/category.validation';
+
+const productRoutes = express.Router();
+productRoutes.post(
+  '/products',
```

Comment:
- Here the path will be only `/`, then in file `.../routes/index.js`, you can specify the main endpoint.

Hint: `routes.use('/products',productRoutes)`

- Implement the same in the category